benchmark/Streamly/Benchmark/Data/Array.hs

```diff
@@ -176,5 +212,6 @@ main = do
             , o_1_space_elimination size
             , o_1_space_transformation size
             , o_1_space_transformationX4 size
+            , o_n_heap_serial size
```

Comment:
We keep benchmarks with different space complexity separate. This should go into a different bench group. See other benchmark modules for examples.